File: src/lib/workout.ts
// Definicja typów dla treningu
export interface WorkoutActivity {
  id: string
  type: "exercise" | "rest"
  name: string
  duration: number | null
  quantity: number | null
  unit: "time" | "reps"
  currentSet: number
  totalSets: number
  sectionName: string
  sectionId: number
  exerciseId: number
}

export interface Workout {
  id: number
  title: string
  description: string | null
  created_at: string
  duration: number | null
  sections: {
    id?: number
    work_id?: number
    name: string
    position: number
    exercises?: {
      id?: number
      ex_id: number
      sets: number
      quantity: number | null
      unit: "CZAS" | "ILOŚĆ"
      duration: number | null
      rest: number
      position: number
      name?: string
    }[]
  }[]
}

File: api/schemas/workout.py
from pydantic import BaseModel
from typing import List, Optional
import datetime
from enum import Enum

class ExerciseUnit(str, Enum):
    TIME = "CZAS"
    QUANTITY = "ILOŚĆ"

class WorkoutExerciseBase(BaseModel):
    ex_id: int
    sets: int
    quantity: Optional[int] = None
    unit: ExerciseUnit
    duration: Optional[int] = None
    rest: int
    position: int

class WorkoutExerciseCreate(WorkoutExerciseBase):
    pass

class WorkoutExerciseResponse(WorkoutExerciseBase):
    id: int
    
    class Config:
        from_attributes = True

class WorkoutSectionBase(BaseModel):
    name: str
    position: int

class WorkoutSectionCreate(WorkoutSectionBase):
    exercises: List[WorkoutExerciseCreate] = []

class WorkoutSectionResponse(WorkoutSectionBase):
    id: int
    work_id: int
    exercises: List[WorkoutExerciseResponse] = []
    
    class Config:
        from_attributes = True

class WorkoutBase(BaseModel):
    title: str
    description: Optional[str] = None
    duration: Optional[int] = None

class WorkoutCreate(WorkoutBase):
    sections: List[WorkoutSectionCreate]

class WorkoutResponse(WorkoutBase):
    id: int
    created_at: datetime.date
    sections: List[WorkoutSectionResponse]
    
    class Config:
        from_attributes = True


